vortexor/src/cli.rs

```diff
@@ -51,6 +52,26 @@ fn get_default_tpu_coalesce_ms() -> &'static str {
     coalesce
 }
 
+/// returns a parser which can validate input URL based on specified schemes.
+fn parse_url_with_scheme(expected_schemes: &'static [&'static str]) -> ValueParser {
```

Comment:
out of scope for this PR:

should these parsing fns be moved to our clap utility libraries? this fn, as well as several others seem to be pretty generic and could be useful in non-vortexor apps we write?

Reply:
Yes. Currently our main code is based on clap v2 -- this code is based on clap v4. When we update to v4, this code should be refactored to some common utilities.